homeassistant/components/vesync/fan.py

```diff
@@ -154,11 +164,18 @@ def set_percentage(self, percentage):
         if not self.smartfan.is_on:
             self.smartfan.turn_on()
 
-        self.smartfan.manual_mode()
-        self.smartfan.change_fan_speed(
-            math.ceil(percentage_to_ranged_value(SPEED_RANGE, percentage))
-        )
-        self.schedule_update_ha_state()
+        try:
+            self.smartfan.manual_mode()
+
+            self.smartfan.change_fan_speed(
+                math.ceil(percentage_to_ranged_value(SPEED_RANGE[self.device.device_type], percentage))
+            )
+            self.schedule_update_ha_state()
+        except:
```

Comment:
Bare exceptions are not allowed. You need to specify the exception that you are expecting.